### PR TITLE
Plugin Recaptcha and Mootools on Joomla 3.4

### PR DESCRIPTION
**Pull request by Qlimax90**
Hi everyone,

in /plugins/captcha/recaptcha/recaptcha.php from line 55 to 79, is there any reason why the new recaptcha is loading the mootools library? Especially on line 76 Mootools gets loaded by JHtml::_('script', $file, true, true);

https://github.com/joomla/joomla-cms/issues/6212

### CODE REVIEW

File: plugins/captcha/recaptcha/recaptcha.php [hunk 1]
```diff
@@ -59,8 +59,6 @@ public function onInit($id = 'dynamic_recaptcha_1')
 
 				$file = $app->isSSLConnection() ? 'https' : 'http';
 				$file .= '://www.google.com/recaptcha/api/js/recaptcha_ajax.js';
-				JHtml::_('script', $file);
```
Comment (mbabker) [inline plugins/captcha/recaptcha/recaptcha.php]: Removing this line completely means you aren't loading the recaptcha_ajax.js file.  FWIW this call isn't loading MooTools.

**Comment by losedk** [comment]
Tested. All good

**Comment by C-Lodder** [comment]
The `addScriptDeclaration` is being injected before the `addScript` statement so would this not lead to undefined variables?

**Comment by losedk** [comment]
Is that a problem? addScript always injects into head

**Comment by dgt41** [comment]
@C-Lodder It would be nice to follow the actual chain of events, first the file and then the initialization :+1:

**Comment by Qlimax90** [comment]
hmm but this would cause an redundancy in the code :-1:

File: plugins/captcha/recaptcha/recaptcha.php [hunk 2]
```diff
@@ -72,15 +70,13 @@ public function onInit($id = 'dynamic_recaptcha_1')
 				$file = $app->isSSLConnection() ? 'https' : 'http';
 				$file .= '://www.google.com/recaptcha/api.js?hl=' . JFactory::getLanguage()
 						->getTag() . '&onload=onloadCallback&render=explicit';
-
-				JHtml::_('script', $file, true, true);
```
Comment (dgt41) [inline plugins/captcha/recaptcha/recaptcha.php]:
```
JHtml::_('script', $file, false, true);
```

File: plugins/captcha/recaptcha/recaptcha.php (hunk 1 above)
Comment (dgt41) [inline plugins/captcha/recaptcha/recaptcha.php]:
```
JHtml::_('script', $file, false, true);
```
This code will not load mootools and also keeps the loading chain untouched, also see the comment below

Comment (dgt41) [inline plugins/captcha/recaptcha/recaptcha.php]: Also using `addScript()` instead of `JHtml::_()` disables the overriding feature of the script file

**Comment by wilsonge** [comment]
@C-Lodder The inline scripts are always added after the files so it's not an issue. That's why we load jQuery no captcha as a file rather than a script

File: plugins/captcha/recaptcha/recaptcha.php (hunk 2 above)
Comment (wilsonge) [inline plugins/captcha/recaptcha/recaptcha.php]: We don't want relative file paths - this is the google script - which isn't stored in Joomla ;)

Comment (dgt41) [inline plugins/captcha/recaptcha/recaptcha.php]: That’s my bad! Of course there is no way to override an external file 😜

**Comment by nueckman** [comment]
Looks good

**Comment by anibalsanchez** [comment]
@test It works Ok, no mootools.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6218">issues.joomla.org/joomla-cms/6218</a>.</sub>

**Comment by zero-24** [comment]
@dgt41 @Qlimax90  what is the status here? Can we leave it now `as it is` or need we to change something? As we get multible successful tests here :smile: Thanks.

**Comment by Qlimax90** [comment]
I think we can leave it as it is :)

**Comment by dgt41** [comment]
@Qlimax90 just do a merge on staging to resolve the conflicts

**Comment by Qlimax90** [comment]
@dgt41 It seems that i have not the rights to do this?
"This pull request contains merge conflicts that must be resolved.
Only those with write access to this repository can merge pull requests."

**Comment by dgt41** [comment]
@Qlimax90 if you pull rebase to latest staging in your local copy, solve the conflicts and push the code back to your online copy, everything should be ok!

**Comment by Kubik-Rubik** [comment]
@Qlimax90 Will you update your PR or should I recreate it? I think that it is a good change (get rid of the loading of the Mootools framework). If you are still keen to do it, then please do it in the next days if possible! If you do it, then please also add a blank line after the switch statement. Thank you in advance!

**Comment by Qlimax90** [comment]
@dgt41 @Kubik-Rubik Please feel free to recreate, at the moment I am struggeling with the local copy due to the fact that I created the PR online. The local branch is not working, sorry...

**Comment by Kubik-Rubik** [comment]
Closed in favor of #7403. Thank you @Qlimax90 for your contribution!

**Pull request closed without merging**